data/generalhw_scripts/gadget-control.py

```diff
@@ -0,0 +1,403 @@
+    '=': (0, 0x2E),               '+': (2, 0x2E),
+    '[': (0, 0x2F),               '{': (2, 0x2F),
+    ']': (0, 0x30),               '}': (2, 0x30),
+    '\\': (0, 0x31),              '|': (2, 0x31),
+    #'#': (0, 0x32),              '~': (2, 0x32),
+    ';': (0, 0x33),               ':': (2, 0x33),
+    '\'': (0, 0x34),              '"': (2, 0x34),
+    '`': (0, 0x35),               '~': (2, 0x35),
+    ',': (0, 0x36),               '<': (2, 0x36),
+    '.': (0, 0x37),               '>': (2, 0x37),
+    '/': (0, 0x38),               '?': (2, 0x38),
+    #'caps lock': (0, 0x39),
+    'f1': (0, 0x3A),
+    'f2': (0, 0x3B),
+    'f3': (0, 0x3C),
+    'f4': (0, 0x3D),
+    'f5': (0, 0x3E),
+    'f6': (0, 0x3F),
+    'f7': (0, 0x40),
+    'f8': (0, 0x41),
+    'f9': (0, 0x42),
+    'f10': (0, 0x43),
+    'f11': (0, 0x44),
+    'f12': (0, 0x45),
+    'printscreen': (0, 0x46),
+    'scroll': (0, 0x47),
+    'pause': (0, 0x48),
+    'insert': (0, 0x49),
+    'home': (0, 0x4A),
+    'pageup': (0, 0x4B),
+    'delete': (0, 0x4C),
+    'end': (0, 0x4D),
+    'pagedown': (0, 0x4E),
+    'right': (0, 0x4F),
+    'left': (0, 0x50),
+    'down': (0, 0x51),
+    'up': (0, 0x52),
+    'ctrl': (1, 0x0),
+    'shift': (2, 0x0),
+    'alt': (4, 0x0),
+    'meta': (8, 0x0),
+    'rctrl': (16, 0x0),
+    'rshift': (32, 0x0),
+    'ralt': (64, 0x0),
+    'rmeta': (128, 0x0),
+}
+
+
+def parse_and_send_cmd(gadget, line):
+    if line.startswith('mouse_'):
+        cmd, param = line.split(' ', 1)
+        if cmd == 'mouse_move':
+            x, y = [int(p) for p in param.split(' ')]
+            #x, y = (32767/1024*x), (32767/768*y)
+            gadget.write_mouse_move_report(int(x), int(y))
+        elif cmd == 'mouse_button':
+            param = int(param)
+            gadget.write_mouse_btn_report(param)
+        else:
+            print("Unknown command: {}".format(cmd), file=sys.stderr)
+        return
+
+    key = line
+    mods = 0
+    while True:
+        if key.startswith('ctrl-'):
+            mods |= 0x01
+            key = key[len('ctrl-'):]
+        elif key.startswith('shift-'):
+            mods |= 0x02
+            key = key[len('shift-'):]
+        elif key.startswith('alt-'):
+            mods |= 0x04
+            key = key[len('alt-'):]
+        else:
+            break
+
+    if key not in keycodes:
+        print("Unknown key: {}".format(key))
+        return
+
+    extra_mods, keycode = keycodes[key]
+    mods |= extra_mods
+    gadget.write_keyboard_report(mods, keycode)
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--storage', default=None, action='store')
+    parser.add_argument('--is-cdrom', default=False, action='store_true')
+    parser.add_argument('--storage-only', default=False, action='store_true')
+    args = parser.parse_args()
+
+    with Gadget(storage_path=args.storage, cdrom=args.is_cdrom, hid=not args.storage_only) as gadget:
+        for line in fileinput.input(files=[]):
```

Comment:
Why not just something like this?
```suggestion
        for line in sys.stdin:
```